### PR TITLE
[NCL-7601] Translate artifact path to identifier instead of the identifier regex to path regex in RepoUtils#removeExcludedArtifacts

### PR DESCRIPTION
**Pull request by patrikk0123**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [x] Have you added unit tests for your change?

### CODE REVIEW

**Comment by patrikk0123** [comment]
The original problem was with the translation of `excludeArtifacts` regex in Maven artifact format to regex in file path format, which was not fully equivalent.

For example, this regex `'.*:.*:.*:.*:(?!db-scripts).*'` => `.*/.*/.*/.*-.*-(?!db-scripts).*\..*$`
the nonpresence of `db-scripts` is checked at the wrong place because of the `.*` and multiple `-` that may be in the input string

Therefore, I did it the other way around by translating the artifact path to the identifier.

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java [hunk 1]
```diff
@@ -344,41 +345,36 @@ public static void recursivelyDeleteEmptyFolder(File element) {
...
-    public static String convertMavenIdentifierToPathRegex(String identifier) {
-        String[] sections = identifier.split(":");
-        if (sections.length != 4 && sections.length != 5) {
-            throw new IllegalArgumentException(
-                    identifier
-                            + " is wrongly formatted! It should be <group-id>:<artifact-id>:<packaging>:<version>:[classifier]");
-        }
-        String fileSeparator = System.getProperty("file.separator");
+    public static String convertArtifactPathToIdentifier(String artifactPath) {
+        var fileSeparator = System.getProperty("file.separator");
```
Comment (jomrazek) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java:357]: Please change this to `File.separator` , we tend to use that when accessing the default separator on other places, also by doing so you can get rid of this variable and use it directly on next line.

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java [hunk 2]
```diff
@@ -243,19 +244,19 @@ public static void removeCommunityArtifacts(File element) {
...
-                Pattern matchesRegex = Pattern.compile(excludedPath);
-                if (matchesRegex.matcher(element.getAbsolutePath()).find()) {
+            for (String exclusionRegex : excludedArtifacts) {
+                Pattern matchesRegex = Pattern.compile(exclusionRegex);
+                var artifactIdentifier = convertArtifactPathToIdentifier(element.getAbsolutePath());
+
+                if (matchesRegex.matcher(artifactIdentifier).find()) {
                     log.debug(
-                            "Removing path {} from the repository since it matches regex: {}",
+                            "Removing path {} from the repository since it the artifact {} matches regex: {}",
```
Comment (jomrazek) [inline pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java:256]: 'it' is obsolete now?

**Review by jomrazek — COMMENTED** [review]
Not a fan of using `var` , small comments, otherwise looks good to me.

File: pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java (hunk 1 above)
Comment (patrikk0123) [reply pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java:357]: I had to do it like this: `Pattern.quote(File.separator)`
Because on Windows, file separator is a `\` which is not a valid regex

**Comment by thescouser89** [comment]
LGTM! Converting `removeExcludedArtifacts` to an identifier to apply the regex. + unit tests for safety, and to handle corner cases in the future.

**Review by thescouser89 — APPROVED** [review]
